File: server/express1/contentType.js
const express = require('express');
const contentTypeRouter = express.Router()
const bodyParser = require('body-parser');
// const multer = require('multer');
// const upload = multer();

contentTypeRouter.use(bodyParser.json()); // for parsing application/json
contentTypeRouter.use(bodyParser.urlencoded({ extended: true })); // for parsing application/x-www-form-urlencoded
// contentTypeRouter.use(upload.array()); // for parsing multipart/form-data

contentTypeRouter.post('/content-type', function (req, res) {
    try {
        let contentType = req.headers['content-type'];
        let urlParams = req.query;
        let data = {};
        if (contentType.includes('text/plain')) {
            let rawData = '';
            req.on('data', (chunk) => { rawData += chunk; });
            req.on('end', () => {
                try {
                    data = JSON.parse(rawData);
                    res.json({ "body": data, 'urlParams': urlParams });
                } catch (e) {
                    console.error(e);
                    res.status(500).send({ "msg": "error", 'code': 0 });
                }
            });
        } else {
            if (req.is('json')) {  // 判断是否为JSON数据
                data = req.body;
            } else if (contentType.includes('application/x-www-form-urlencoded')) {
                data = JSON.parse(Object.keys(req.body)[0]);
            } else if (contentType.includes('multipart/form-data')) {
                data = JSON.parse(req.body);
            } else {
                data = req.body;
            }
            res.json({ "body": data, 'urlParams': urlParams });
        }
    } catch (e) {
        console.error(e);
        res.status(500).send({ "msg": "error", 'code': 0 });
    }
});

contentTypeRouter.get('/content-type', function (req, res) {
    res.json({ "msg": "GET request received" });
});

module.exports = contentTypeRouter

File: server/python/contenttype.py
from flask import request, jsonify
from flask import Blueprint
import json

contentTypeRouter = Blueprint("contenttype",__name__,url_prefix="/ct")
@contentTypeRouter.route("/content-type", methods=['GET', 'POST'])
def applicationJson():
    if request.method == 'POST':
        try:
            contentType=request.headers.get('Content-Type')
            urlParams=request.args.to_dict()
            data={}
            # if(contentType=='application/json'): #这种写法不好，因为1，有些请求头的值是application/json;charset=UTF-8，2，有点header是小写的，3，有些请求的body部分不是有效的json
            if(request.is_json): #is_json包含了application/json和text/json,application/json;charset=UTF-8;boundary=xxx等等，并且body部分是有效的json
                data = request.get_json()
                return jsonify({"body":data,'urlParams':urlParams})
            elif(contentType=='application/x-www-form-urlencoded'):
                data = request.form.to_dict()
                data_str = list(data.keys())[0]
                data = json.loads(data_str)
            elif(contentType=='multipart/form-data'):
                data = request.form.to_dict()
                data = json.loads(request.data.decode())
            elif(contentType=='text/plain'):
                data = json.loads(request.data.decode())
            else:
                data = request.data
            return jsonify({"body":data,'urlParams':urlParams})
        except Exception as e:
            print(e)
            return jsonify({"msg": "error",'code':0})
    elif request.method == 'GET':
        return jsonify({"msg": "GET request received"})